openapi/openapi.yaml

```diff
@@ -764,48 +1016,150 @@ components:
-                has started.
-            state:
-              type: string
-              example: terminated
+                Datetime in ISO format informing about when the inspection has
+                started.
+            pods:
+              type: array
+              items:
+                type: object
+                description: Status information for the Pods created by the inspection job.
+                additionalProperties: false
+                required:
+                  - container
+                  - exit_code
+                  - finished_at
+                  - reason
+                  - started_at
+                  - state
+                properties:
+                  container:
+                    type: string
+                    description: SHA of container image in which the inspection is done.
+                    nullable: true
+                  exit_code:
+                    type: integer
+                    description: >
+                      Return code of the process performing inspection (user supplied
+                      script return value).
+                    nullable: true
+                  finished_at:
+                    type: string
+                    description: >
+                      Datetime in ISO format informing about when the inspection has
+                      finished.
+                    nullable: true
+                  reason:
+                    type: string
+                    description: Reasoning on finished inspection run.
+                    nullable: true
+                  started_at:
+                    type: string
+                    description: >
+                      Datetime in ISO format informing about when the inspection has
+                      started.
+                  state:
+                    type: string
+                    example: terminated
+        workflow:
+          type: object
+          description: Status of the submitted inspection Workflow.
+          additionalProperties: true
+    InspectionWorkflowParameters:
+      type: object
+      description: Inspection Workflow parameters.
+      required:
+        - dockerfile
+        - inspection-id
+        - specification
+      properties:
+        dockerfile:
+          type: string
+          description: Dockerfile.
+        inspection-id:
+          type: string
+          description: ID of the inspection.
+        specification:
+          type: string
+          description: Inspection specification.
+        batch-size:
+          type: string
+          default: '1'
+          example: '1'
+          description: Size of the batch.
+        batch-name:
+          type: string
+          description: Name of the batch.
+        allowed-failures:
+          type: string
+          default: '1'
+          example: '1'
+          description: Number of allowed failures.
+        parallelism:
+          type: string
+          default: '1'
```

Comment:
Are these string?

Reply:
Ye, it's quite counterintuitive, but otherwise, the resource is rejected.

Reply:
They can cause troubles during syncs to the database where they will need to be mapped to integer columns. Can they be cast on the server-side to make endpoint type clean?

Reply:
Right, sure, we can do that similarly as we did for the `batch_size` :)

Reply:
Ah no wait... we already do that for parallelism as well (I did think it was weird that I skipped the parallelism and implemented it for the rest of them):
https://github.com/CermakM/amun-api/blob/d254af1417ba562d6020ab1737a51f91ba15d84c/amun/api_v1.py#L130

Reply:
In the specification, it is of `integer` type: https://github.com/CermakM/amun-api/blob/d254af1417ba562d6020ab1737a51f91ba15d84c/openapi/openapi.yaml#L465
But the workflow has to receive a string.

Reply:
> But the workflow has to receive a string.

This one is used in `InspectionResponse` model. Can API cast these from string to int for API consumer before sending them? API can internally still operate on strings as needed, but for the outside world, it will use ints when talking to API or retrieving response from endpoints.

Reply:
That would create an inconsistency between what is actually send to the workflow-controller and what we display to users. I.e., when you take those workflow parameters returned by the InspectionResponse, I think you should be able to use them in the workflow directly.

What is the reasoning behind this? What's stored into the database is only the inspection specification, where the strings are mapped correctly to integers, isn't it?

Reply:
> That would create an inconsistency between what is actually send to the workflow-controller and what we display to users. I.e., when you take those workflow parameters returned by the InspectionResponse, I think you should be able to use them in the workflow directly.

Amun is a service that runs inspections - the fact it uses argo for this and that this casting is necessary is a detail that does not need to be exposed to users. Actually, if I'm a Amun service consumer, I would expect these values being int, because they are actually integers.

`InspectionSpecification` expects an integer, but the model returned back using `InspectionResponse` keeps strings. I think this is an inconsistency from API endpoint pov. The same is propagated to amun client.

> What is the reasoning behind this? What's stored into the database is only the inspection specification, where the strings are mapped correctly to integers, isn't it?

Yes, as of now. But we want to have these data.

Reply:
Aight, that sounds reasonable. I'll see whether we can do some processing before returning the InspectionResponse as well :)